src/main/kotlin/frc/robot/autonomous/AutonomousCommands.kt

```diff
@@ -0,0 +1,112 @@
+package frc.robot.autonomous
+
+import com.pathplanner.lib.auto.AutoBuilder
+import com.pathplanner.lib.path.PathPlannerPath
+import edu.wpi.first.math.kinematics.ChassisSpeeds
+import edu.wpi.first.wpilibj2.command.Command
+import edu.wpi.first.wpilibj2.command.Commands
+import edu.wpi.first.wpilibj2.command.button.Trigger
+import frc.robot.lib.flipIfNeeded
+import frc.robot.subsystems.feeder
+import frc.robot.swerveDrive
+
+fun feederPath(pathName: String, mirror: Boolean = false): Command =
+    AutoBuilder.followPath(
+            if (mirror) PathPlannerPath.fromPathFile(pathName).mirrorPath()
+            else PathPlannerPath.fromPathFile(pathName)
+        )
+        .andThen(
+            Commands.run({
+                swerveDrive.runVelocity(ChassisSpeeds(0.8, 0.0, 0.0))
+            })
+        )
+        .raceWith(feeder(Trigger { true }))
+
+fun B1L(): Command =
+    Commands.sequence(
+        AutoBuilder.resetOdom(
+            PathPlannerPath.fromPathFile("B1L").startingHolonomicPose.get()
+        ),
+        AutoBuilder.followPath(PathPlannerPath.fromPathFile("B1L")),
+        Commands.runOnce({ selectedScorePose = { Reef1Left.flipIfNeeded() } }),
+        autoScoreL4()
+    )
+
+fun B1R(): Command =
+    Commands.sequence(
+        AutoBuilder.resetOdom(
+            PathPlannerPath.fromPathFile("B1R").startingHolonomicPose.get()
+        ),
+        AutoBuilder.followPath(PathPlannerPath.fromPathFile("B1R")),
+        Commands.runOnce({ selectedScorePose = { Reef1Right.flipIfNeeded() } }),
+        autoScoreL4()
+    )
+
+fun C6L(): Command =
+    Commands.sequence(
+        AutoBuilder.resetOdom(
+            PathPlannerPath.fromPathFile("C6L").startingHolonomicPose.get()
+        ),
+        AutoBuilder.followPath(PathPlannerPath.fromPathFile("C6L")),
+        Commands.runOnce({ selectedScorePose = { Reef6Left.flipIfNeeded() } }),
+        autoScoreL4()
+    )
+
+private fun S5L(): Command =
+    Commands.sequence(
+        AutoBuilder.followPath(PathPlannerPath.fromPathFile("S5L")),
+        Commands.runOnce({ selectedScorePose = { Reef5Left.flipIfNeeded() } }),
+        autoScoreL4()
+    )
+
+private fun S5R(): Command =
+    Commands.sequence(
+        AutoBuilder.followPath(PathPlannerPath.fromPathFile("S5R")),
+        Commands.runOnce({ selectedScorePose = { Reef5Right.flipIfNeeded() } }),
+        autoScoreL4()
+    )
```

Comment:
These commands are pretty identical. Extract the logic to a private function that takes parameters. Whether to reset the odometry or not should also be a parameter.